multicluster-config/observer-dev-values.yaml

```diff
@@ -0,0 +1,107 @@
+# kind create cluster --name observer-cluster
+# kind create cluster --name workload-cluster1
+
+thanosStorage:
+  secret:
+    name: thanos-objstore-config
+    key: objstore.yml
+  # More details about supported storage types and config options can be found here: https://thanos.io/tip/thanos/storage.md/
+  config: ""
+  # config: |-
+  #   type: S3
+  #   config:
+  #     endpoint: "{{ include "thanos.minio.fullname" . }}.{{ .Release.Namespace }}.svc.cluster.local:9000"
+  #     bucket: "thanos"
+  #     access_key: "minio"
+  #     secret_key: "minio123"
+  #     insecure: true
+
+thanos:
+  enabled: true
+  existingObjstoreSecret: thanos-objstore-config
+  queryFrontend:
+    enabled: true
+  query:
+    replicaLabel:
+      - prometheus_replica
+    # logLevel: debug
+    dnsDiscovery:
+      sidecarsService: kube-prometheus-thanos-discovery
+      sidecarsNamespace: "{{ .Release.Namespace }}"
+    # URLs of exposed Thanos query components on remote workload clusters
+    stores: []
+  ruler:
+    enabled: true
+    # logLevel: debug
+    replicaLabel: ruler_replica
+    clusterName: observer
+    alertmanagers:
+      - http://dns+alertmanager-operated:9093
+    # TODO: Investigate how to properly configure rules for thanos ruler at scale
```

Comment:
prometheus rules need to be moved to thanos ruler

Reply:
**Discussion:** Where alerts should be evaluated?

- **On the central cluster** - Thanos ruler
  - Pros:
    - ruler is able to use all data components (buckets/prometheus instances)
    - alerts are centralized - one Alertmanager
  - Cons:
    - There is no automatic discovery of ruler in Ruler, we would need to use some other tool (simlar to [this](https://github.com/observatorium/thanos-rule-syncer) ) or extend jsonnet translator functionality
    - alerts are centralized -> decreased reliability, according to [this article](https://www.robustperception.io/federation-what-is-it-good-for) alerts should be as deep down in hierarchy as possible  

- **On the workload clusters** - Prometheus
  - Pros:
    - increased reliability (e.g. in case of network problem between workload and observer)
    - automatic discovery of rules
  - Cons:
    - need to store enough data in prometheus PVCs to evaluate all alerts (can't use thanos bucket storage)
    - need to process alerts from multiple replicas of prometheus in HA mode

**How others do it**
- [thanos ruler website](https://thanos.io/tip/components/rule.md/#rule-aka-ruler): _It is recommended to keep deploying rules inside the relevant Prometheus servers locally. Use ruler only on specific cases._ (Because of reliability.)
- [particule.io](https://particule.io/en/blog/thanos-monitoring/): Alertmanager at each cluster, no Ruler.
- [mattermost](https://mattermost.com/blog/monitoring-cloud-environments-at-scale-with-prometheus-and-thanos/): Alertmanager + Ruler only at observer cluster.
- [improbable.io](https://www.improbable.io/blog/thanos-architecture-at-improbable) : same as mattermost